current_ranges_test.cpp

```diff
@@ -0,0 +1,28 @@
+#include "current_ranges.h"
+#include <stdio.h>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+#include <assert.h>
+
+int main ()
+{
+	
+	// Posistive : Number of Ranges in the Current Readings
+	std::vector<CurrentReadingsParameters*> RangeAndReadings = {};
+	processCurrentReadingsData({ 1, 2, 2, 3, 4, 24, 25, 26 }, RangeAndReadings);
+	assert(RangeAndReadings.size()==2);
+	
+	// Positive : Check for Minimum, Maximum values and total number of readings of ranges
+	processCurrentReadingsData({ 1, 2, 2, 3, 4, 24, 25, 26 }, RangeAndReadings);
```

Comment:
Very nice test - I could clearly see the call and the expected return from it.
Suggestion: You are re-using the same variable (RangeAndReadings) across two tests. This may cause interference between them. Better to isolate each test in its own function and use local variables. That way the tests stay independent.